clippy_lints/src/utils/author.rs

```diff
@@ -333,6 +334,24 @@ impl<'a, 'tcx> PrintVisitor<'a, 'tcx> {
 
     #[allow(clippy::too_many_lines)]
     fn expr(&self, expr: &Binding<&hir::Expr<'_>>) {
+        if let Some(macro_call) = root_macro_call_first_node(self.cx, expr.value)
+            && is_format_macro(self.cx, macro_call.def_id)
+        {
+            let diag = self.cx.tcx.item_name(macro_call.def_id);
+            bind!(self, macro_call);
+            chain!(self, "let Some({macro_call}) = macros::root_macro_call_first_node(cx, {expr})");
+            chain!(self, "cx.tcx.is_diagnostic_item(sym::{diag}_macro, {macro_call}.def_id)");
```

Comment:
I like `sym::{diag}_macro`, and it *almost* works, for exception of `panic`, it should be `std_panic_macro`, but this implementation returns just `panic_macro`. I would appreciate a better solution here, but if there's none, I will hardcode this specific case, although it seems more like a workaround.

Reply:
Would `tcx.get_diagnostic_name` work?

Reply:
Seems to work like a charm, thanks!